src/librustc_trans/mir/constant.rs

```diff
@@ -96,11 +96,11 @@ impl<'tcx> Const<'tcx> {
         let llty = type_of::type_of(ccx, ty);
         let val = match cv {
             ConstVal::Float(v) => {
-                let v_f64 = match v {
-                    ConstFloat::F32(v) => f32::from_bits(v) as f64,
-                    ConstFloat::F64(v) => f64::from_bits(v)
+                let bits = match v {
+                    ConstFloat::F32(v) => U32(v),
+                    ConstFloat::F64(v) => U64(v)
                 };
-                C_floating_f64(v_f64, llty)
+                consts::bitcast(Const::from_constint(ccx, &bits).llval, llty)
```

Comment:
@alexcrichton Was there a concern at some point that passing signaling NaNs to LLVM could trigger an exception on the host? Mostly wondering in terms of marking it as a solved problem.